### DISCUSSION TITLE
How to achieve image scaling and adaptive width

### DISCUSSION THREAD

**Discussion by imxood** [emilk/egui#927] [Q&A] [answered]
I want to realize that when the window becomes wider or narrower, the picture can adapt to the window, what should I do?

`egui::Window` `egui::Image`

![image](https://user-images.githubusercontent.com/14890680/144443372-565df9ed-f6f0-494b-b62a-690eed446924.png)

**Comment by coderedart** [comment, marked as answer]
Did you try just passing the  https://docs.rs/egui/0.15.0/egui/struct.Ui.html#method.available_size as the size for image widget?
Just choose whatever container like egui area or central panel and inside the closure get the remaining area to set as image widget size.

**Reply by imxood** [reply]
It is great~~
```rust
let response = Image::new(texture, ui.available_size()).sense(Sense::drag()).ui(ui);
```
![image](https://user-images.githubusercontent.com/14890680/144527481-cd78fc6b-b2b7-48d9-8eec-8ba2a64554e9.png)

**Comment by emilk** [comment]
The http example app does this:

``` rust
                    let mut size = egui::Vec2::new(image.size[0] as f32, image.size[1] as f32);
                    size *= (ui.available_width() / size.x).min(1.0);
                    ui.image(texture_id, size);
```

This gets a proportional scaling, if you are interested in that.

**Reply by imxood** [reply]
it's good
